deps/wazuh_testing/wazuh_testing/remote.py

```diff
@@ -0,0 +1,136 @@
+    """
+    msg = fr"Started \(pid: \d+\). Listening on port {port}\/{protocol} \({connection_type}\)."
+
+    return make_callback(pattern=msg, prefix=REMOTED_DETECTOR_PREFIX)
+
+
+def callback_detect_syslog_event(message):
+    """Creates a callback to detect the syslog messages in the archives.log
+
+    Args:
+        message (str): syslog message sent through the socket
+
+    Returns:
+        callable: callback to detect this event
+    """
+    expr = fr".*->\d+\.\d+\.\d+\.\d+\s{message}"
+    return make_callback(pattern=expr, prefix=None)
+
+
+def send_syslog_message(message, port, protocol, manager_address="127.0.0.1"):
+    """This function sends the ping message to the manager
+
+    This message is the first of many between the manager and the agents. It is used to check if both of them are ready
+    to send and receive other messages
+
+    Args:
+        message (str): string to send as a syslog event.
+        protocol (str): it can be UDP or TCP.
+        port (int): port where the manager has bound the remoted port
+        manager_address (str): address of the manager.
+
+    Raises:
+        ConnectionRefusedError: if there's a problem while sending messages to the manager
+    """
+    if protocol == "UDP":
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    else:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    if not message.endswith("\n"):
+        message += "\n"
+
+    sock.connect((manager_address, port))
+    sock.send(message.encode())
+    sock.close()
+
+
+def create_archives_log_monitor():
+    """Creates a FileMonitor for the archives.log file
+
+    Returns:
+        FileMonitor: object to monitor the archives.log
+    """
+    # Reset ossec.log and start a new monitor
+    truncate_file(ARCHIVES_LOG_FILE_PATH)
+    wazuh_archives_log_monitor = FileMonitor(ARCHIVES_LOG_FILE_PATH)
+
+    return wazuh_archives_log_monitor
+
+
+def detect_archives_log_event(archives_monitor, callback, error_message, update_position=True, timeout=5):
+    archives_monitor.start(timeout=timeout, update_position=update_position, callback=callback,
+                           error_message=error_message)
+
+
+def check_syslog_event(wazuh_archives_log_monitor, message, port, protocol):
+    """Check if a syslog event is properly receive by the manager.
+
+    Args:
+        wazuh_archives_log_monitor (FileMonitor): FileMonitor object to monitor the archives.log.
+        message (str): Message sent for syslog that must appear in the archives.log.
+        protocol (str): it can be UDP or TCP.
+        port (int): port where the manager has bound the remoted port
+    """
+    send_syslog_message(message, port, protocol)
+    detect_archives_log_event(archives_monitor=wazuh_archives_log_monitor,
+                              callback=callback_detect_syslog_event(message),
+                              error_message="Syslog message wasn't received or took too much time.")
+
```

Comment:
The function names may cause some confusion. For now, all events have been searched for in the ossec.log, and particularly this function searches for them in the archives.log. We have to think of something to know with the name where they are being looked for. In this case something like `check_syslog_event_in_archives`

Reply:
This function can be used to search for any event in the `archives.log` where many of the events received by remoted are logged. That's why there's no point to change the name to `check_syslog_event_in_archives` when it can be used to search any event. 

In the future, we should add a new module to `wazuh_testing` for searching events in the log that can be used by any other module, however, this is no the time to do it.